tests/configlet/util/generic_patch.py

```diff
@@ -0,0 +1,217 @@
+        if ((e["path"].split("/")[1] in filter) and
+                (not re.search(rm_pattern, e["path"]))):
+            jpatch.append(e)
+
+    if not hack_apply:
+        patch_file = os.path.join(patch_dir, "patch_0.json")
+        with open(patch_file, "w") as s:
+            s.write(json.dumps(jpatch, indent=4))
+    else:
+        # Hack:
+        # Split Interface off of tghe patch into seperate file
+        # Have this until generic patch updater will order 
+        # Interfaces add/remove correctly. Also remove port-admin-up
+        # as that should come at the end.
+        #
+        add_intf = False
+        interface_patch = []
+        other_patch = []
+
+        intf_pattern = ("^/INTERFACE/Ethernet[0-9][0-9]*$"
+                "|"
+                "^/INTERFACE/Ethernet[0-9][0-9]*\\|[0-9a-fA-F][0-9a-fA-F\\.:]*[0-9a-fA-F]~1[0-9][0-9]*$")
+
+        admin_up_pattern = "^/PORT/Ethernet[0-9][0-9]*/admin_status"
+
+        for e in jpatch:
+            if re.search(intf_pattern, e["path"]):
+                interface_patch.append(e)
+                add_intf = e["op"] == "add"
+            elif not re.search(admin_up_pattern, e["path"]):
+                other_patch.append(e)
+
+        patch_intf_file = os.path.join(patch_dir,
+                "patch_{}_intf.json".format(0 if add_intf else 1))
+        patch_other_file = os.path.join(patch_dir,
+                "patch_{}_other.json".format(1 if add_intf else 0))
+
+        with open(patch_intf_file, "w") as s:
+            s.write(json.dumps(interface_patch, indent=4))
+
+        with open(patch_other_file, "w") as s:
+            s.write(json.dumps(other_patch, indent=4))
+
+
+def _list_patch_files(patch_dir):
+    return sorted(fnmatch.filter(os.listdir(patch_dir), "patch_[0-9]_*.json"))
+
+
+def generic_patch_add_t0(duthost, skip_load=False, hack_apply=False):
+    # Load config w/o T0
+    #
+    if not skip_load:
+        duthost.copy(src=os.path.join(no_t0_db_dir, "config_db.json"),
+                dest="/etc/sonic/config_db.json")
+        config_reload(duthost, wait=RELOAD_WAIT_TIME, start_bgp=True)
+
+    if hack_apply:
+        # Hack: TODO: Before adding port, patch updater need to ensure
+        # the port is down. Until then bring it down explicitly.
+        #
+        tor_ifname = tor_data["links"][0]["local"]["sonic_name"]
+        duthost.shell("config interface shutdown {}".format(tor_ifname))
+        do_pause(PAUSE_INTF_DOWN,
+                "pause upon i/f {} shutdown before add patch".format(tor_ifname))
+
+    patch_files = _list_patch_files(patch_add_t0_dir)
+
+    CMD = CMD_APPLY_HACK if hack_apply else CMD_APPLY
+
+    for fl in patch_files:
+        sonic_fl = os.path.join("/etc/sonic", fl)
+        duthost.copy(src=os.path.join(patch_add_t0_dir, fl), dest=sonic_fl)
+        res = duthost.shell(CMD.format(sonic_fl), module_ignore_errors=True)
+
+        # HACK: TODO: There are scenarios, where it applies patch and still consider as failed
```

Comment:
Do we have tracking for this issue?

Reply:
It is part of AddRack issue.
When you fix it, running & passing this test code manually in switch will be required to stamp it as "done".
At that point, you will be able to figure out specifics.